### PR TITLE
feature/node-delete

### LINKED ISSUE #5492: Allow to delete a node from web admin with a locationlog opened
**Is your feature request related to a problem? Please describe.**
Currently, the only way to delete a node is to rely on pfmon node_cleanup task (using last_seen attribute).

When you delete a node from web admin, API checks if a locationlog is opened for a node which is always the case once we received a RADIUS request for a node. A locationlog is always closed to open a new one.

**Describe the solution you'd like**
Be able to delete a node from web admin **even** if a locationlog is opened (using a bypass mechanism and a warning message).

**Additional context**
Recent code has been added to disallow deletion of node in web admin like we had with `pfcmd` (see #5088)

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Cleanup node before deletion

NEWS file entries
=======================
Enhancements
-------------
* Allow to delete a node from web admin with a locationlog opened (#5492)

Issue
=====
fixes #5492

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue [hunk 1]
```diff
@@ -51,6 +51,10 @@
               <icon class="position-absolute mt-1" name="project-diagram"></icon>
               <span class="ml-4">{{ $t('Apply Bypass VLAN') }}</span>
             </b-dropdown-item>
+            <b-dropdown-item @click="applyBulkDelete()">
+              <icon class="position-absolute mt-1" name="trash-alt"></icon>
+              <span class="ml-4">{{ $t('Delete') }}</span>
```
Comment (julsemaan) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue]: I'm really not sure about allowing the ability to bulk delete.

This has the potential to cause a sh*tload of problems.

I'm fine with leaving the API call there and the admin plumbing but I'd remove the option for now

**Review by julsemaan — CHANGES_REQUESTED** [review]

**Comment by extrafu** [comment]
@satkunas Please remove the bulk option from the GUI. @cgx have a quick look at the JS. @julsemaan merge after please

File: html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js [hunk 2]
```diff
@@ -393,6 +393,15 @@ const actions = {
     }).catch(err => {
       commit('NODE_ERROR', err.response)
     })
+  },
+  bulkDelete: ({ commit }, data) => {
+    commit('NODE_REQUEST')
+    return api.bulkDelete(data).then(response => {
+      commit('NODE_BULK_SUCCESS', response)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js]: We need a new mutation (`NODE_BULK_DELETED`), which does the same thing as `NODE_DESTROYED` and `NODE_NOT_EXISTS` but for the list of deleted nodes.

Comment (satkunas) [reply html/pfappserver/root/static.alt/src/views/Nodes/_store/index.js]: My last commit hid the bulk delete, so there's no longer any need for this method in the store module.

**Comment by cgx** [comment]
@julsemaan I think we're ready to merge.

**Pull request merged**